packages/server/lib/browsers/firefox-util.ts

```diff
@@ -0,0 +1,249 @@
+import Bluebird from 'bluebird'
+import Debug from 'debug'
+import _ from 'lodash'
+import Marionette from 'marionette-client'
+import Exception from 'marionette-client/lib/marionette/error'
+import { Command } from 'marionette-client/lib/marionette/message.js'
+import util from 'util'
+import Foxdriver from '@benmalka/foxdriver'
+import protocol from './protocol'
+
+const debug = Debug('cypress:server:browsers:firefox-util')
+
+const promisify = (fn) => {
+  return (...args) => {
+    return new Bluebird((resolve, reject) => {
+      fn(...args, (data) => {
+        if ('error' in data) {
+          reject(new Exception(data, data))
+        } else {
+          resolve(data)
+        }
+      })
+    })
+  }
+}
+
+let forceGcCc: () => Promise<void>
+
+let timings = {
+  gc: [] as any[],
+  cc: [] as any[],
+  collections: [] as any[],
+}
+
+const logGcDetails = () => {
```

Comment:
should we expose this to the user somehow? currently it is only used by e2e test `1_firefox_spec`